src/python/perfflowaspect/advice_chrome.py

```diff
@@ -260,11 +260,9 @@ def around(func):
         @functools.wraps(func)
         def trace(*args, **kwargs):
             event = ChromeTracingAdvice.__create_event_from_func(func)
-            event["ph"] = "B"
-            ChromeTracingAdvice.__flush_log(json.dumps(event) + ",")
+            event["ph"] = "X"
```

Comment:
Minor nit: 

In the commit message, please split the title and the body by adding a empty line between two.

Generally, we recommend to wrap the body text at 72 columns. 

Also please add more details about what `X` does compared to the old scheme and why it reduces trace amounts etc.

Reply:
I would also like to see some comments about the switch from X to B, I might have missed some of the earlier discussions.

Reply:
@slabasan I added the comments to the commit. The basic idea is to combine B and E events into X events so that we don't duplicate data in the log.

Reply:
I haven't looked too much into this part of the code yet, so please feel free to correct me. My understanding is that two of the types of events (ph) are `B`, for "before", and `E`, for "after". If I copy two entries from the pfw output file (one for each type of event) here:
```
{"name": "bas", "cat": "__main__", "pid": 26449, "tid": 4619326976, "ts": 1638916424816701.0, "ph": "B"}
{"name": "bas", "cat": "__main__", "pid": 26449, "tid": 4619326976, "ts": 1638916424816794.8, "ph": "E"}
```
I see all of the fields are duplicated, except for `ts`. If I understand your code change, the pfw file will now have a single entry for `B` or `E` events, that is denoted as `X`, and it will have a new key called `dur` instead of `ts`. Couple questions that I have:
- In the current scheme, is there always a matching `E` event for each `B` event?
- Is it possible to derive the original `B` and `E` from `X`, or should we provide a parameter to let the user switch between the two types of outputs?
- Also I notice the change is only in the python annotations, should we also make these changes in the c annotations as well either in this PR or a new PR?

Reply:
> I haven't looked too much into this part of the code yet, so please feel free to add things I have missed. My understanding is that two of the types of events (ph) are `B`, for "before", and `E`, for "after". If I copy two entries from the pfw output file (one for each type of event) here:
> 
> ```
> {"name": "bas", "cat": "__main__", "pid": 26449, "tid": 4619326976, "ts": 1638916424816701.0, "ph": "B"}
> {"name": "bas", "cat": "__main__", "pid": 26449, "tid": 4619326976, "ts": 1638916424816794.8, "ph": "E"}
> ```
> 
> I see all of the fields are duplicated, except for `ts`. If I understand your code change, the pfw file will now have a single entry for `B` or `E` events, that is denoted as `X`, and it will have a new key called `dur` instead of `ts`. Couple questions that I have:

Yes, based on my understanding of the format, we should use B and E events when we don't have the start and end together as in let say we start the timer out of scope of the function and stop it somewhere else. 

For only before and only after events we should probably use instant events [doc](https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview#heading=h.lenwiilchoxp). This wont change the size of log but is represented differently on chrome tracing visualization. 

For events which have both B and E, the X event is a more compact format to represent those. 

> 
> * In the current scheme, is there always a matching `E` event for each `B` event?

I guess if we only have before event or only end event, I am not sure how it will be shown on the timeline. As I explained we should use instant events for that.

> * Is it possible to derive the original `B` and `E` from `X`, or should we provide a parameter to let the user switch between the two types of outputs?

Yes, we can derive. however, based on my understanding this is not a user-facing thing. Either way, the user will see the same timeline output. The point here is to represent the same around events more efficient.

> * Also I notice the change is only in the python annotations, should we also make these changes in the c annotations as well either in this PR or a new PR?

Either way is fine with me.

Reply:
Ah thank you for sharing the [trace format doc](https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview#heading=h.lenwiilchoxp)! I didn't realize there was a standard defined format, we should make a note to add these to the our PerfFlow docs as well.

We might want to have a discussion on being able to switch between compact and verbose format in the trace files. If you can show that the change does not break any major functionality, is it useful to have such an option? @dongahn Can you think of some use cases for having a toggle to switch between `X` and `B`/`E` events?

Regarding adding this change to the c annotations, I think if the change is small, we should include it in this PR as well.

Reply:
I have to believe there are reasons why both tracing options are available and making this configurable makes a lot of sense. Another control that we can introduce in the future into our TOML support? To begin with for this PR, it makes sense using other ways to toggle (envVar?)

Reply:
Makes sense to provide a switch between the two types of events, and yes we could easily support this as an environment variable first, and eventually provide it as a config in TOML.

Reply:
Ok, do you guys have a convention for env variables? I was thinking PERFFLOWASPECT_EVENT=COMPACT/VERBOSE.

Reply:
Yes, to give you an idea on the convention, here are the available options:
```
PERFFLOW_OPTIONS="log-dir=./logdir"
PERFFLOW_OPTIONS="log-filename-include=hostname,instance-path,pid,name"
PERFFLOW_OPTIONS="name=mycomponent:log-filename-include=name"
PERFFLOW_OPTIONS="log-enable=False"  <-- currently being added in a PR 38
```

Reply:
So it will be PERFFLOW_OPTIONS="log-event=compact/verbose"